src/Lyn.Protocol/Common/Messages/NetworkMessageSerializer.cs

```diff
@@ -0,0 +1,55 @@
+using System;
+using System.Buffers;
+using Lyn.Protocol.Bolt1.Messages;
+using Lyn.Protocol.Bolt1.TlvStreams;
+using Lyn.Types.Serialization;
+
+namespace Lyn.Protocol.Common.Messages
+{
+    public class NetworkMessageSerializer<TMessage> : INetworkMessageSerializer
```

Comment:
ah nice, this works with the DI like that? (with T is only used in the constructor)